Sources/Workspace/Workspace.swift

```diff
@@ -326,13 +327,26 @@ public class Workspace {
         self.artifactsPath = self.dataPath.appending(component: "artifacts")
         self.cachePath = cachePath
 
-        self.containerProvider = RepositoryPackageContainerProvider(
-            repositoryManager: repositoryManager,
-            mirrors: self.config.mirrors,
-            manifestLoader: manifestLoader,
-            currentToolsVersion: currentToolsVersion,
-            toolsVersionLoader: toolsVersionLoader
-        )
+        if enablePackageRegistry {
+            PackageModel._useLegacyIdentities = false
```

Comment:
is setting this here safe? could code run before this point use different identities, etc?